lib/pdn/pdn.c

```diff
@@ -619,15 +620,31 @@ int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
 	fmt = "+CGCONTRDP: %*u,,\"%*[^\"]\",\"\",\"\",\"%15[0-9.]\",\"%15[0-9.]\",,,,,%u";
 
 	/* If IPv4 is enabled, it will be the first response line. */
-	ret = nrf_modem_at_scanf(at_cmd, fmt, &dns4_pri_str, &dns4_sec_str, ipv4_mtu);
-	if (ret != 3) {
+	matched = nrf_modem_at_scanf(at_cmd, fmt, &dns4_pri_str, &dns4_sec_str, &mtu);
+	/* Need to match at least the two IP addresses, or there is an error */
+	if (matched < 2) {
```

Comment:
If `matched == 2`, is it guaranteed that it is the two IP addresses that matched, and not the the first IP address + the MTU?

Reply:
yes :)